### PR TITLE
Upgrade to react 18

### PR DESCRIPTION
**Pull request by JGoggers**
#### Changes made

Upgraded packages so the plugin does not break with upgrades to react 18.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,5 +61,8 @@
       "prettier --write"
     ]
   },
-  "packageManager": "yarn@3.6.0"
+  "packageManager": "yarn@3.6.0",
+  "dependencies": {
+    "@backstage/plugin-catalog-react": "^1.9.2"
```
Comment (drodil) [inline package.json:66]: Why is this a dependency?

**Comment by drodil** [comment]
Thanks, one comment though

**Comment by JGoggers** [comment]
Thanks! When will this be released?

**Comment by drodil** [comment]
@JGoggers I try to release it today

**Comment by acierto** [comment]
@drodil in `main` resolutions still overrides it to react 17. It has to be or bumped up to react 18 or removed (if it can work without resolution section)

**Comment by drodil** [comment]
@acierto good catch. Will take a look tomorrow if nobody has time to make a PR before that!

**Comment by fshowalter** [comment]
This should have been a major release no? By resolving to 18, you've dropped support for 17 and will trigger peer fails on anyone not on 18 yet.

**Comment by drodil** [comment]
The problem here is that only MUIv4 does not work with React18 so TSC will fail with 18 as devDep in root package.json. If I understand this correctly, the resolution only affects the local dev environment and not the packaged plugin. We should either use React17 still in the local env or bump also the MUI to 5 (though not sure what is the plan in backstage for this)

**Comment by drodil** [comment]
Apparently, the transition to MUI5 is still in very much progress and all plugins in backstage are still using MUI5. The only place the MUI5 is used is the theme package. So maybe it's coming at some point in time but we should not hold our breath for that.

**Comment by fshowalter** [comment]
Yeah, it's a tough spot. But the resolution (while intended for Dev here) triggers warnings in production. Do a simple create app with backstage on 17, add the plugin, then do yarn install --production and you'll see the peer warning that toolbox wants 18. I wonder how the core plugins are avoiding the problem...

**Pull request merged**